ray-curator/tests/test_get_ray_client.py

```diff
@@ -0,0 +1,106 @@
+
+# -----------------------------------------------------------------------------
+# Helpers
+# -----------------------------------------------------------------------------
+
+METRIC_NAME = "ray_gcs_actors_count"
+
+
+def _query_prometheus(prometheus_port: int, metric_name: str = METRIC_NAME) -> list[dict]:
+    """Query Prometheus for a specific metric and return the raw results list."""
+    url = f"http://localhost:{prometheus_port}/api/v1/query"
+    response = requests.get(url, params={"query": metric_name}, timeout=5)
+    response.raise_for_status()
+    payload = response.json()
+    if payload["status"] != "success":  # pragma: no cover
+        return []
+    return payload["data"].get("result", [])
+
+
+def _wait_for_metric(session_name: str, prometheus_port: int, timeout: int = 60) -> bool:
+    """Poll Prometheus until the metric for *session_name* appears or timeout."""
+    deadline = time.time() + timeout
+    while time.time() < deadline:
+        try:
+            results = _query_prometheus(prometheus_port)
+        except requests.RequestException:
+            # Prometheus may not be reachable yet, wait and retry.
+            results = []
+        for result in results:
+            if result.get("metric", {}).get("SessionName") == session_name:
+                return True
+        time.sleep(2)
+    return False
+
+
+def _latest_session_name(ray_temp_dir: str, timeout: int = 30) -> str:
+    """Return the latest Ray session directory name inside *ray_temp_dir*."""
+    session_link = os.path.join(ray_temp_dir, "session_latest")
+    deadline = time.time() + timeout
+    while time.time() < deadline:
+        if os.path.islink(session_link):
+            target = os.readlink(session_link)
+            return os.path.basename(target.rstrip(os.sep))
+        time.sleep(1)
+    msg = "Ray session directory not created in time."
+    raise RuntimeError(msg)
+
+
+# -----------------------------------------------------------------------------
+# Tests
+# -----------------------------------------------------------------------------
+
+
+def test_single_cluster_metrics_visible():
+    """Start one Ray cluster and ensure its metrics appear in Prometheus."""
+    # Clear RAY_ADDRESS
+    os.environ.pop("RAY_ADDRESS", None)
+    with tempfile.TemporaryDirectory(prefix="ray_test_single_") as ray_tmp:
+        client_info = get_ray_client(ray_temp_dir=ray_tmp)
+        prometheus_port = client_info["prometheus_web_port"] or 9090  # Fallback if already running
+
+        session_name = _latest_session_name(ray_tmp)
+        assert _wait_for_metric(session_name, prometheus_port), (
+            f"Metric for session {session_name} not found in Prometheus on port {prometheus_port}."
+        )
+
+    # TODO: This will kill all the ray processes. Find a better way to do this.
+    # Cleanup : stop any Ray processes we started
+    subprocess.run(["ray", "stop", "--force"], check=False, capture_output=True)  # noqa: S603, S607
```

Comment:
yup this todo might be important.. would hate for it to kill my existing job if i end up running it on my cluster while i have the job.. i had to hack it in conftest  in this PR too https://github.com/NVIDIA-NeMo/Curator/pull/844

Reply:
Hmm, any good solution for this? In the case of getting a client, the ray init happens in a subprocess, so it's not straightforward to do a shutdown on it?